Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -1198,11 +1200,30 @@ WarpXOpenPMDPlot::WriteOpenPMDFieldsAll ( //const std::string& filename,
                 auto const chunk_offset = getReversedVec( box_offset );
                 auto const chunk_size = getReversedVec( local_box.size() );
 
-                amrex::Real const * local_data = fab.dataPtr( icomp );
-                mesh_comp.storeChunk( openPMD::shareRaw(local_data),
-                                      chunk_offset, chunk_size );
+                // we avoid relying on managed memory by copying explicitly to host
+                //   remove the copies and "streamSynchronize" if you like to pass
+                //   GPU pointers to the I/O library
+
+#ifdef AMREX_USE_GPU
+                if (fab.arena()->isManaged() || fab.arena()->isDevice()) {
+                    std::shared_ptr<amrex::Real> data_pinned(
+                        (amrex::Real*)amrex::The_Pinned_Arena()->alloc(local_box.numPts()*sizeof(amrex::Real)),
+                        amrex::DataDeleter{amrex::The_Pinned_Arena()});
```

Comment:
@WeiqunZhang mentioned: There is another possibility.
```suggestion
                    amrex::BaseFab<amrex::Real> foo(local_box, 1, amrex::The_Pinned_Arena());
                    std::shared_ptr<amrex::Real> data_pinned(foo.release());
```
But wouldn't that still need the custom deleter? Maybe it copies it when constructed with a unique_ptr...?

Reply:
Yes, the `shared_ptr` will get the deleter from the `unique_ptr`.